samples/sample-07/src/test/java/com/example/sample07/Sample07ApplicationTest.java

```diff
@@ -0,0 +1,335 @@
+		final Resource resource = new ClassPathResource(BROKER_PROPERTIES_FILE_PATH);
+		try {
+			final Properties properties = PropertiesLoaderUtils.loadAllProperties(resource.getFilename());
+			return properties.stringPropertyNames().stream()
+							.collect(Collectors.toMap(
+										s -> s,
+										s -> (String) properties.get(s)));
+		}
+		catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@EnableKafka
+	@TestConfiguration
+	public static class TestConfig {
+
+		CountDownLatch partitionRevokedLatch = new CountDownLatch(0);
+
+		CountDownLatch partitionAssignedLatch = new CountDownLatch(1);
+
+		CountDownLatch listenerLatch = new CountDownLatch(2);
+
+		@KafkaListener(id = GROUP_ID, topics = TOPIC_NAME, autoStartup = "false")
+		void listen(ConsumerRecord<Integer, String> ignored) {
+			listenerLatch.countDown();
+		}
+
+		@Bean
+		KafkaTemplate<Integer, String> template(ProducerFactory<Integer, String> producerFactory) {
+			return new KafkaTemplate<>(producerFactory);
+		}
+
+		@Bean
+		ProducerFactory<Integer, String> producerFactory() {
+			return new DefaultKafkaProducerFactory<>(KafkaTestUtils.producerProps(BROKER));
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<Integer, String>
+		kafkaListenerContainerFactory(ConsumerFactory<Integer, String> consumerFactory) {
+
+			final ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+					new ConcurrentKafkaListenerContainerFactory<>();
+
+			factory.setConsumerFactory(consumerFactory);
+
+			factory.getContainerProperties().setConsumerRebalanceListener(new ConsumerAwareRebalanceListener() {
+				@Override
+				public void onPartitionsAssigned(Consumer<?, ?> consumer,
+												Collection<TopicPartition> partitions) {
+					partitionAssignedLatch.countDown();
+				}
+
+				@Override
+				public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+					partitionRevokedLatch.countDown();
+				}
+			});
+			return factory;
+		}
+
+		@Bean
+		ConsumerFactory<Integer, String> consumerFactory() throws InterruptedException {
+			Properties propsDummy = new Properties();
+			propsDummy.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, BytesDeserializer.class.getName());
+			propsDummy.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+									StringDeserializer.class.getName());
+			propsDummy.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BROKER);
+			propsDummy.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "hello");
+			propsDummy.setProperty(ConsumerConfig.GROUP_PROTOCOL_CONFIG, "consumer");
+			propsDummy.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+			Consumer<Object, Object> dummyConsumer = new KafkaConsumer<>(propsDummy);
+
+			Thread.sleep(5000);
+			dummyConsumer.close();
+
+			final Map<String, Object> props = getConsumerProperties();
+			return new DefaultKafkaConsumerFactory<>(props);
+		}
+
+		@Bean
+		public ThreadPoolTaskExecutor threadPoolTaskExecutor() {
```

Comment:
We don't need this bean. Spring Boot auto-configures for us one: `TaskExecutionAutoConfiguration`.
If we need some custom its properties, they can be set from the `application.properties`: `spring.task.execution` prefix.